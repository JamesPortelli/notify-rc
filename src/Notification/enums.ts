export enum Kind {
  Success,
  Warning,
  Error,
  Info
}

export enum Position {
  TopLeft,
  TopRight,
  BottomLeft,
  BottomRight,
  Top,
  Bottom,
  Centre
}
