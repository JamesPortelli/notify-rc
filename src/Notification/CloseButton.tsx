import React from 'react';
import styled, { css } from 'styled-components';

import { Position } from './enums';

interface Props {
  position: Position;
  onClick(): void;
}

type ButtonProps = Pick<Props, 'position'>;

const Button = styled.div<ButtonProps>`
  margin: 10px;
  line-height: 0;

  > svg {
    cursor: pointer;
  }

  ${props =>
    props.position !== Position.Top && props.position !== Position.Bottom
      ? css`
          text-align: right;
        `
      : css`
          float: right;
        `}
`;

export function CloseButton({ position, onClick }: Props) {
  return (
    <Button position={position}>
      <svg data-testid="close-button" width="18" height="18" onClick={onClick}>
        <line x1="5" y1="13" x2="13" y2="5" stroke="white" strokeWidth="2" />
        <line x1="5" y1="5" x2="13" y2="13" stroke="white" strokeWidth="2" />
      </svg>
    </Button>
  );
}
