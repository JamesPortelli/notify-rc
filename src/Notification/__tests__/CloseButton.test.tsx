import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';

import { CloseButton } from '../CloseButton';
import { Position } from '../enums';

const setup = (propsOverride = {}) => {
  const props = {
    position: Position.Top,
    onClick: jest.fn(),
    ...propsOverride
  };

  return {
    props,
    ...render(<CloseButton {...props} />)
  };
};

describe('CloseButton Tests', () => {
  describe('when position is TOP or Bottom', () => {
    it('will render a CloseButton floated to the right', () => {
      const { container } = setup();

      expect(container).toMatchSnapshot();
    });
  });

  describe('when position is not TOP or Bottom', () => {
    it('will render a CloseButton aligned to the right', () => {
      const { container } = setup({ position: Position.TopRight });

      expect(container).toMatchSnapshot();
    });
  });

  describe('when the close button is clicked', () => {
    it('will call the callback function', () => {
      const { getByTestId, props } = setup({ position: Position.TopRight });

      act(() => {
        fireEvent.click(getByTestId('close-button'));
      });

      expect(props.onClick).toHaveBeenCalled();
    });
  });
});
