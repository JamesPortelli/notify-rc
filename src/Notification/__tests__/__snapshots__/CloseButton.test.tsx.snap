// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton Tests when position is TOP or Bottom will render a CloseButton floated to the right 1`] = `
.c0 {
  margin: 10px;
  line-height: 0;
  float: right;
}

.c0 > svg {
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <svg
      data-testid="close-button"
      height="18"
      width="18"
    >
      <line
        stroke="white"
        stroke-width="2"
        x1="5"
        x2="13"
        y1="13"
        y2="5"
      />
      <line
        stroke="white"
        stroke-width="2"
        x1="5"
        x2="13"
        y1="5"
        y2="13"
      />
    </svg>
  </div>
</div>
`;

exports[`CloseButton Tests when position is not TOP or Bottom will render a CloseButton aligned to the right 1`] = `
.c0 {
  margin: 10px;
  line-height: 0;
  text-align: right;
}

.c0 > svg {
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <svg
      data-testid="close-button"
      height="18"
      width="18"
    >
      <line
        stroke="white"
        stroke-width="2"
        x1="5"
        x2="13"
        y1="13"
        y2="5"
      />
      <line
        stroke="white"
        stroke-width="2"
        x1="5"
        x2="13"
        y1="5"
        y2="13"
      />
    </svg>
  </div>
</div>
`;
